python_ta/debug/snapshot.py

```diff
@@ -15,6 +34,9 @@ def snapshot():
     frame = inspect.currentframe().f_back
 
     while frame:
+        if len(local_vars) == 0:  # to ensure we do not add main stack frame multiple times
```

Comment:
This still isn't what I'm looking for. Look at the condition on line 40 and ask why it's there, and how you can use it to address the current problem.